src/robot_manager.cpp

```diff
@@ -753,7 +784,14 @@ catch(resource_registrar::TemotoErrorStack& e)
 bool RobotManager::goalNavigationCb(RobotNavigationGoal::Request& req, RobotNavigationGoal::Response& res)
 try
 {
+  TEMOTO_INFO_("Received a goal Navigation request");
+  TEMOTO_DEBUG_STREAM_("Request:\n" << req);
+
+  // I have commented out this line because I need to cancel the goal. If Uncommented, cancel cmd is triggered after sendGoal() finishes its execution
```

Comment:
refer to my previous comment on why this is happening. this mutex is important, as the feature is subject to data race because of asynchronous `sendGoal`, `cancelGoal`, and `feedback` functions